File: usedesk_sdk/src/main/java/ru/usedesk/sdk/data/framework/loader/TokenLoader.java
package ru.usedesk.sdk.data.framework.loader;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import javax.inject.Inject;
import javax.inject.Singleton;

import ru.usedesk.sdk.data.repository.user.info.DataLoader;

@Singleton
public class TokenLoader extends DataLoader<String> {
    private static final String PREF_NAME = "usedeskSdkToken";
    private static final String KEY_TOKEN = "token";//TODO: replace

    private final SharedPreferences sharedPreferences;

    @Inject
    public TokenLoader(@NonNull Context context) {
        this.sharedPreferences = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }

    @Nullable
    @Override
    protected String loadData() {
        return sharedPreferences.getString(KEY_TOKEN, null);
    }

    @Override
    protected void saveData(@NonNull String token) {
        sharedPreferences.edit()
                .putString(KEY_TOKEN, token)
                .apply();
    }

    @Override
    public void clearData() {
        super.clearData();

        sharedPreferences.edit()
                .remove(KEY_TOKEN)
                .apply();
    }
}

File: chat-sdk/src/main/java/ru/usedesk/chat_sdk/data/repository/configuration/loader/token/TokenLoader.kt
package ru.usedesk.chat_sdk.data.repository.configuration.loader.token

import android.content.Context
import android.content.SharedPreferences
import ru.usedesk.chat_sdk.data.repository._extra.DataLoader

internal class TokenLoader(
    context: Context
) : DataLoader<String>(), ITokenLoader {

    private val sharedPreferences: SharedPreferences = context.getSharedPreferences(
        PREF_NAME,
        Context.MODE_PRIVATE
    )

    override fun loadData() = sharedPreferences.getString(KEY_TOKEN, null)

    override fun saveData(data: String) {
        sharedPreferences.edit()
            .putString(KEY_TOKEN, data)
            .apply()
    }

    override fun clearData() {
        sharedPreferences.edit()
            .remove(KEY_TOKEN)
            .apply()
    }

    companion object {
        private const val PREF_NAME = "usedeskSdkToken"
        private const val KEY_TOKEN = "token"
    }
}